packages/smooth_app/lib/widgets/screen_visibility.dart

```diff
@@ -25,10 +25,16 @@ class _ScreenVisibilityDetectorState extends State<ScreenVisibilityDetector> {
 
   @override
   Widget build(BuildContext context) {
+    // The first time the Widget is called, [onVisibilityChanged] is not called
+    // And it will keep the initial value -> [false] which is wrong
+    if (!_notifier.isVisible) {
+      _notifier.updateValue(true);
+    }
+
```

Comment:
I can understand the logic but the code is a bit confusing.

What about:
```dart
bool firstTime = true;

Widget build(context) {
   if (firstTime) {
      firstTime = false;
      _notifier.updateValue(true);
   }
```

Or just `final ScreenVisibility _notifier = ScreenVisibility(true);`

That said, I'm not sure if the fact that a widget is being build means that it is visible (e.g. with the 3 tabbed pages), but that's another story.

Reply:
It's a state less widget, so a first time variable can not work.